modules/openapi-generator/src/main/resources/dart/class.mustache

```diff
@@ -58,14 +58,24 @@ class {{classname}} {
     return {
     {{#vars}}
       {{#isDateTime}}
-      '{{baseName}}': {{name}} == null ? '' : {{name}}.toUtc().toIso8601String(){{^-last}},{{/-last}}
+        '{{baseName}}': {{name}} == null ? '' : {{name}}.toUtc().toIso8601String(){{^-last}},{{/-last}}
       {{/isDateTime}}
       {{#isDate}}
-      '{{baseName}}': {{name}} == null ? '' : {{name}}.toUtc().toIso8601String(){{^-last}},{{/-last}}
+        '{{baseName}}': {{name}} == null ? '' : {{name}}.toUtc().toIso8601String(){{^-last}},{{/-last}}
       {{/isDate}}
       {{^isDateTime}}
       {{^isDate}}
-      '{{baseName}}': {{name}}{{^-last}},{{/-last}}
+        {{#complexType}}
+          {{#isListContainer}}
+            '{{baseName}}': {{name}} == null ? null : {{name}}.map((item) => item.toJson()).toList(){{^-last}},{{/-last}}
+          {{/isListContainer}}
+          {{^isListContainer}}
+            '{{baseName}}': {{name}} == null ? null : {{name}}.toJson(){{^-last}},{{/-last}}
+          {{/isListContainer}}
+        {{/complexType}}
+        {{^complexType}}
+          '{{baseName}}': {{name}}{{^-last}},{{/-last}}
+        {{/complexType}}
```

Comment:
@agilob Can we keep the indentation as 2-space? It would make the code review a lot easier.

Reply:
This MR will cause a bunch of conflicts with 2 other MR I have in progress and it's not my changes, I resurrected commit from 
@arndt-s. I will take care of it as the last thing if you dont mind?

Reply:
Agreed with taking care of the code indentation in another PR.